stock_dropshipping_whole_supplier/models/sale.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2018 Akretion (<http://www.akretion.com>)
+# @author: Florian da Costa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _action_confirm(self):
+        for order in self:
+            order._set_dropship_route_on_adhoc_cases()
+        super()._action_confirm()
+
+    def _set_dropship_route_on_adhoc_cases(self):
+        """ Assign a dropship route to sale lines which have
+            the same vendor than dropship product
+        """
+        self.ensure_one()
+        vendor_dropsh, vendor_line = self._get_dropship_info()
+        for vendor, line in vendor_dropsh.items():
+            lines_to_dropshipping = vendor_line.get(vendor)
+            if lines_to_dropshipping:
+                lines_to_dropshipping.write(
```

Comment:
I'd prefer here building a recordset and then write the value(as it is the same). It will reduces sql UPDATEs

Reply:
Done